pkg/build/prune/resolvers.go

```diff
@@ -98,30 +98,34 @@ func (o *perBuildConfigResolver) Resolve() ([]*buildapi.Build, error) {
 	completeStates := util.NewStringSet(string(buildapi.BuildPhaseComplete))
 	failedStates := util.NewStringSet(string(buildapi.BuildPhaseFailed), string(buildapi.BuildPhaseError), string(buildapi.BuildPhaseCancelled))
 
-	results := []*buildapi.Build{}
+	prunableBuilds := []buildapi.Build{}
```

Comment:
You're right. I went originally with `ByCreationTimestamp` as `[]Build` following the definition of `BuildList`:
```go
// BuildList is a collection of Builds.
type BuildList struct {
	kapi.TypeMeta
	kapi.ListMeta

	// Items is a list of builds
	Items []Build
}
```

It's more reasonable to have `ByCreationTimestamp` as `[]*Build` and do explicity type conversions where needed.